django/middleware/csrf.py

```diff
@@ -160,23 +161,52 @@ def _reject(self, request, reason):
         )
         return _get_failure_view()(request, reason=reason)
 
-    def process_view(self, request, callback, callback_args, callback_kwargs):
-        if getattr(request, 'csrf_processing_done', False):
-            return None
-
+    def _get_current_token(self, request):
         try:
-            cookie_token = request.COOKIES[settings.CSRF_COOKIE_NAME]
+            if settings.CSRF_USE_SESSIONS:
+                if not getattr(request, 'session', None):
+                    raise ImproperlyConfigured(
+                        'No session could be found for the request, but CSRF_USE_SETTINGS is '
+                        'turned on.'
+                    )
+                session_token = request.session[settings.CSRF_SESSION_KEY]
```

Comment:
Merge with the line below since you are not using `session_token` later on. Is it actually needed to sanitize the session token? After all the user cannot change it.